Sources/Datadog/RUM/RUMMonitor/Scopes/RUMResourceScope.swift

```diff
@@ -127,44 +119,44 @@ internal class RUMResourceScope: RUMScope {
         }
         let resourceType: RUMResourceType = resourceKindBasedOnRequest ?? command.kind
 
-        let eventData = RUMResourceEvent(
+        let resourceEvent = RUMResourceEvent(
             dd: .init(
                 browserSdkVersion: nil,
                 session: .init(plan: .plan1),
                 spanId: spanId,
                 traceId: traceId
             ),
-            action: context.activeUserActionID.flatMap { rumUUID in
+            action: self.context.activeUserActionID.map { rumUUID in
                 .init(id: rumUUID.toRUMDataFormat)
             },
-            application: .init(id: context.rumApplicationID),
+            application: .init(id: self.context.rumApplicationID),
             ciTest: dependencies.ciTest,
-            connectivity: dependencies.connectivityInfoProvider.current,
+            connectivity: .init(context: context),
             context: .init(contextInfo: attributes),
-            date: dateCorrection.applying(to: resourceStartTime).timeIntervalSince1970.toInt64Milliseconds,
+            date: resourceStartTime.timeIntervalSince1970.toInt64Milliseconds,
```

Comment:
The `resourceStartTime` will not include NTP correction if it's taken from [`URLSessionTaskTransactionMetrics`](https://developer.apple.com/documentation/foundation/urlsessiontasktransactionmetrics) (it means auto-instrumented resources):
```swift
/// Metrics values take precedence over other values.
if let metrics = resourceMetrics {
   resourceStartTime = metrics.fetch.start
   resourceDuration = metrics.fetch.end.timeIntervalSince(metrics.fetch.start)
   size = metrics.responseSize ?? command.size
}
```
Hence, using `resourceStartTime` without `dateCorrection.applying(to:)` will lead to skewing "resource to view" temporal relationship (🐞) and eventually even starting a resource before the view is started. Mind that RUM uses nanoseconds precision and NTP correction is typically milliseconds.